nohost/src/HTMLServer.js

```diff
@@ -0,0 +1,154 @@
+        // the path cannot be served
+        if (localPath === this._pathResolver(localPath)) {
+            return false;
+        }
+
+        // Url ending in "/" implies default file, which is usually index.html.
+        // Return true to indicate that we can serve it.
+        if (localPath.match(/\/$/)) {
+            return true;
+        }
+
+        // FUTURE: do a MIME Type lookup on file extension
+        console.log("ans: ", LiveDevelopmentUtils.isStaticHtmlFileExt(localPath));
+        return LiveDevelopmentUtils.isStaticHtmlFileExt(localPath);
+    };
+
+    /**
+     * When a livedocument is added to the server cache, make sure live
+     * instrumentation is enabled
+     */
+    HTMLServer.prototype.add = function (liveDocument) {
+        if (liveDocument.setInstrumentationEnabled) {
+            // enable instrumentation
+            liveDocument.setInstrumentationEnabled(true);
+        }
+        BaseServer.prototype.add.call(this, liveDocument);
+    };
+
+    /**
+     * Serve the contents of a path into the filesystem,
+     * invoking the appropriate content handler, and rewriting any resources
+     * in the local filesystem to Blob URLs.
+     */
+    HTMLServer.prototype.servePath = function(path, callback) {
+        var fs = this.fs;
+
+        fs.stat(path, function(err, stats) {
+            if(err) {
+                return callback(err);
+            }
+
+            // If this is a dir, error
+            if(stats.isDirectory()) {
+                return callback(new Error('expected file path'));
+            }
+
+            // This is a file, pick the right content handler based on extension
+            var ext = Path.extname(path);
+
+            if(Content.isHTML(ext)) {
+                Handlers.handleHTML(path, fs, callback);
+            } else {
+                Handlers.handleFile(path, fs, callback);
+            }
+        });
+    };
+
+    /**
+     * Serve an existing HTML fragment/file (i.e., one that has already been read
+     * for a given path) from the local filesystem, rewriting any resources
+     * in the local filesystem to Blob URLs. The original path of the file is needed
+     * in order to locate other resources with paths relative to this file.
+     */
+    HTMLServer.prototype.serveHTML = function(html, path, callback) {
+        var fs = this.fs;
+
+        Rewriter.rewriteHTML(html, path, fs, function(err, rewrittenHTML) {
+            if(err) {
+                Log.error('unable to rewrite HTML for `' + path + '`');
+                // TODO: best way to deal with error here? 500?
+                return Handlers.handle404(path, callback);
+            }
+
+            callback(null, rewrittenHTML);
+        });
+    };
+
+    /**
+     * If a livedoc exists, serves the instrumented version of the file as as a blob URL.
+     * Otherwise, it serves only the file's contents as a blob URL.
+     */
+    HTMLServer.prototype.maybeServeLiveDoc = function(path, callback) {
```

Comment:
I wonder if the way to do this is to move `_liveDocuments` onto the filesystem blob cache, and generate it at the same time that we write the blob, in the case that it's an HTML file.  That way we can get the BLOB URL in one step without having to do an async read of disk, and move all this code to `pathToUrl`.